qtfred/src/mission/missionsave.cpp

```diff
@@ -3773,6 +3785,13 @@ int CFred_mission_save::save_players()
 
 		fout(")");
 
+		// make sure we have at least one dogfight weapon for each ship type in a dogfight mission
+		if (IS_MISSION_MULTI_DOGFIGHT && (num_dogfight_weapons != Team_data[i].num_ship_choices)) {
+			MessageBox(nullptr,
+				"Warning: This mission is a dogfight mission but no dogfight weapons are available for at least one ship in the loadout!", "No dogfight weapons",
+				MB_OK);
+		}
```

Comment:
Suggestion if you have the skill to pull it off: record the ships that don't have dogfight weapons available and then add them to this warning.  An `SCP_string` would make the process fairly easy.

Reply:
The number of ships this could include is technically unlimited, making the error message potentially incredibly long. Otherwise I'd do it.

Instead it's probably better to print each ship to the log and include that instruction in the message itself.

Reply:
Ah, smart.

Reply:
and done